lisa/sut_orchestrator/baremetal/platform_.py

```diff
@@ -0,0 +1,138 @@
+from lisa.environment import Environment
+from lisa.platform_ import Platform
+from lisa.sut_orchestrator.baremetal.build.source import Source, SourceSchema
+from lisa.sut_orchestrator.baremetal.cluster.cluster import Cluster, ClusterSchema
+from lisa.sut_orchestrator.baremetal.cluster.ip_getter import IpGetterChecker
+from lisa.sut_orchestrator.baremetal.cluster.readychecker import ReadyChecker
+from lisa.util import check_till_timeout, fields_to_dict
+from lisa.util.logger import Logger
+from lisa.util.shell import try_connect
+from lisa.util.subclasses import Factory
+
+from .. import BAREMETAL
+from . import features
+
+
+@dataclass_json()
+@dataclass
+class SourceBuildSchema:
+    source: Optional[SourceSchema] = None
+
+
+@dataclass_json()
+@dataclass
+class BareMetalPlatformSchema:
+    source: Optional[SourceSchema] = field(default=None)
+    cluster: ClusterSchema = field(default_factory=ClusterSchema)
+
+
+class BareMetalPlatform(Platform):
+    def __init__(
+        self,
+        runbook: schema.Platform,
+    ) -> None:
+        super().__init__(runbook=runbook)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return BAREMETAL
+
+    @classmethod
+    def supported_features(cls) -> List[Type[feature.Feature]]:
+        return [features.StartStop]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        baremetal_runbook: BareMetalPlatformSchema = self.runbook.get_extended_runbook(
+            BareMetalPlatformSchema
+        )
+        assert baremetal_runbook, "platform runbook cannot be empty"
+        self._baremetal_runbook = baremetal_runbook
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return True
+
+    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
+        # copy build (shared, check if it's copied)
+        source_path: str = ""
+        if self._baremetal_runbook.source and self._baremetal_runbook.source:
+            source_factory = Factory[Source](Source)
+            source = source_factory.create_by_runbook(self._baremetal_runbook.source)
+            self._log.debug(f"found build '{source.type_name()}', to expand runbook.")
+            source.initialize()
+            source_path = source.source_path
+
+        ready_checker_factory = Factory[ReadyChecker](ReadyChecker)
+        cluster_factory = Factory[Cluster](Cluster)
+        ip_getter_factory = Factory[IpGetterChecker](IpGetterChecker)
+        ip_address: str
+        cluster = self._baremetal_runbook.cluster
+        # ready checker cleanup
+        if cluster.ready_checker:
+            ready_checker_runbook = ready_checker_factory.create_by_runbook(
+                cluster.ready_checker
+            )
+            ready_checker_runbook.clean_up()
+
+        # copy build if source exists
+        cluster_runbook = cluster_factory.create_by_runbook(cluster)
+        cluster_runbook.initialize(source_path=source_path)
```

Comment:
The "copy" step shouldn't be in `_initialize`. It's not easy to understand the logic. So it needs a method called "copy".